pkg/memtier/tracker_idlepage.go

```diff
@@ -154,10 +159,10 @@ func (t *TrackerIdlePage) RemovePids(pids []int) {
 }
 
 func (t *TrackerIdlePage) removePid(pid int) {
-	t.mutex.Lock()
-	defer t.mutex.Unlock()
 	delete(t.regions, pid)
+	t.mutex.Lock()
```

Comment:
Do you think we could handle all locking with the existing mutex, or is there some issue with it? Introducing separate regionsMutex makes it quite a bit more complicated?

Here delete(t.regions, pid) seems to be left without lock protection. But if RemovePids would acquire t.mutex in the first place, it could execute all removals from t.regions and t.accesses without worries.